### ISSUE TITLE
repmgr fails to start postgres after changing follower

### ISSUE THREAD

**Issue by xeago** [bitnami/bitnami-docker-postgresql-repmgr#30]
**Description**

Upon a replica changing follower repmgr fails to start postgres properly.

**Steps to reproduce the issue:**

1. Have a healthy 3 node cluster on k8s
2. Kill the primary
3. Observe the replica failing

**Describe the results you received:**

Killed 1, the current primary, 0 takes over.
```
2020-08-10T13:08:56.743563573Z [2020-08-10 13:08:56] [NOTICE] notifying node "pgr-dev-xeago-postgres-ha-postgresql-2" (ID: 1002) to follow node 1000
2020-08-10T13:08:56.744082868Z INFO:  node 1002 received notification to follow node 1000
2020-08-10T13:08:56.875027527Z [2020-08-10 13:08:56] [NOTICE] monitoring cluster primary "pgr-dev-xeago-postgres-ha-postgresql-0" (ID: 1000)
2020-08-10T13:09:02.899810758Z [2020-08-10 13:09:02] [NOTICE] new standby "pgr-dev-xeago-postgres-ha-postgresql-2" (ID: 1002) has connected
```

Observe 2 dying,
```
2020-08-10T13:21:13.885263493Z NOTICE: stopping server using "/opt/bitnami/postgresql/bin/pg_ctl  -D '/bitnami/postgresql/data' -w -m fast stop"
2020-08-10T13:21:13.885270773Z DEBUG: executing:
2020-08-10T13:21:13.885275392Z   /opt/bitnami/postgresql/bin/pg_ctl  -D '/bitnami/postgresql/data' -w -m fast stop
2020-08-10T13:21:18.996863997Z DEBUG: result of command was 0 (0)
2020-08-10T13:21:18.996901842Z DEBUG: local_command(): output returned was:
2020-08-10T13:21:18.996908805Z waiting for server to shut down......... done
2020-08-10T13:21:18.996914113Z 
2020-08-10T13:21:18.996919023Z NOTICE: starting server using "/opt/bitnami/postgresql/bin/pg_ctl  -w -D '/bitnami/postgresql/data' start"
2020-08-10T13:21:18.996924355Z DEBUG: executing:
2020-08-10T13:21:18.996929212Z   /opt/bitnami/postgresql/bin/pg_ctl  -w -D '/bitnami/postgresql/data' start
2020-08-10T13:21:19.11385252Z DEBUG: result of command was 141 (36096)
2020-08-10T13:21:19.113897746Z DEBUG: local_command(): output returned was:
2020-08-10T13:21:19.113904426Z waiting for server to shut down......... done
2020-08-10T13:21:19.113911997Z waiting for server to start....postgres: could not access the server configuration file "/bitnami/postgresql/data/postgresql.conf": No such file or directory
2020-08-10T13:21:19.113917934Z 
...
2020-08-10T13:21:45.176605249Z WARNING: node "pgr-dev-xeago-postgres-ha-postgresql-2" not found in "pg_stat_replication"
2020-08-10T13:21:45.176639769Z DEBUG: sleeping 27 of max 30 seconds waiting for standby to attach to primary
2020-08-10T13:21:46.17856605Z WARNING: node "pgr-dev-xeago-postgres-ha-postgresql-2" not found in "pg_stat_replication"
2020-08-10T13:21:46.178614762Z DEBUG: sleeping 28 of max 30 seconds waiting for standby to attach to primary
2020-08-10T13:21:47.18061579Z WARNING: node "pgr-dev-xeago-postgres-ha-postgresql-2" not found in "pg_stat_replication"
2020-08-10T13:21:47.180657961Z DEBUG: sleeping 29 of max 30 seconds waiting for standby to attach to primary
2020-08-10T13:21:48.182410652Z WARNING: node "pgr-dev-xeago-postgres-ha-postgresql-2" not found in "pg_stat_replication"
2020-08-10T13:21:48.182444161Z DEBUG: sleeping 30 of max 30 seconds waiting for standby to attach to primary
2020-08-10T13:21:49.182647952Z NOTICE: STANDBY FOLLOW successful
2020-08-10T13:21:49.182693841Z DETAIL: standby attached to upstream node "pgr-dev-xeago-postgres-ha-postgresql-1" (ID: 1001)
2020-08-10T13:21:49.182700923Z DEBUG: _create_event(): event is "standby_follow" for node 1002
2020-08-10T13:21:49.182705998Z DEBUG: get_recovery_type(): SELECT pg_catalog.pg_is_in_recovery()
2020-08-10T13:21:49.183438838Z DEBUG: _create_event():
2020-08-10T13:21:49.183598523Z    INSERT INTO repmgr.events (              node_id,              event,              successful,              details             )       VALUES ($1, $2, $3, $4)    RETURNING event_timestamp 
2020-08-10T13:21:49.185938263Z DEBUG: _create_event(): Event timestamp is "2020-08-10 13:21:49.183604+00"
2020-08-10T13:21:49.185960658Z DEBUG: _create_event(): command is '/opt/bitnami/repmgr/events/router.sh %n %e %s "%t" "%d"'
2020-08-10T13:21:49.186132268Z INFO: executing notification command for event "standby_follow"
2020-08-10T13:21:49.18614618Z DETAIL: command is:
2020-08-10T13:21:49.186151481Z   /opt/bitnami/repmgr/events/router.sh 1002 standby_follow 1 "2020-08-10 13:21:49.183604+00" "standby attached to upstream node \"pgr-dev-xeago-postgres-ha-postgresql-1\" (ID: 1001)"
2020-08-10T13:21:49.33246524Z [2020-08-10 13:21:49] [ERROR] connection to database failed
2020-08-10T13:21:49.33249886Z [2020-08-10 13:21:49] [DETAIL] 
2020-08-10T13:21:49.332503287Z could not connect to server: Connection refused
2020-08-10T13:21:49.33250792Z 	Is the server running on host "pgr-dev-xeago-postgres-ha-postgresql-2.pgr-dev-xeago-postgres-ha-postgresql-headless.pgr-dev.svc.cluster.local" (10.0.14.70) and accepting
2020-08-10T13:21:49.332514218Z 	TCP/IP connections on port 5432?
2020-08-10T13:21:49.332519159Z 
2020-08-10T13:21:49.332523719Z [2020-08-10 13:21:49] [DETAIL] attempted to connect using:
2020-08-10T13:21:49.332528602Z   user=repmgr password=snip connect_timeout=5 dbname=repmgr host=pgr-dev-xeago-postgres-ha-postgresql-2.pgr-dev-xeago-postgres-ha-postgresql-headless.pgr-dev.svc.cluster.local port=5432 fallback_application_name=repmgr
2020-08-10T13:21:50.33350498Z [2020-08-10 13:21:50] [ERROR] connection to database failed
2020-08-10T13:21:50.333549939Z [2020-08-10 13:21:50] [DETAIL] 
2020-08-10T13:21:50.333556647Z could not connect to server: Connection refused
2020-08-10T13:21:50.33356195Z 	Is the server running on host "pgr-dev-xeago-postgres-ha-postgresql-2.pgr-dev-xeago-postgres-ha-postgresql-headless.pgr-dev.svc.cluster.local" (10.0.14.70) and accepting
2020-08-10T13:21:50.33356753Z 	TCP/IP connections on port 5432?
2020-08-10T13:21:50.333572301Z 
2020-08-10T13:21:50.333576901Z [2020-08-10 13:21:50] [DETAIL] attempted to connect using:
2020-08-10T13:21:50.333582071Z   user=repmgr password=snip connect_timeout=5 dbname=repmgr host=pgr-dev-xeago-postgres-ha-postgresql-2.pgr-dev-xeago-postgres-ha-postgresql-headless.pgr-dev.svc.cluster.local port=5432 fallback_application_name=repmgr
2020-08-10T13:21:51.334567361Z [2020-08-10 13:21:51] [ERROR] connection to database failed
2020-08-10T13:21:51.33460567Z [2020-08-10 13:21:51] [DETAIL] 
2020-08-10T13:21:51.334611868Z could not connect to server: Connection refused
2020-08-10T13:21:51.334617142Z 	Is the server running on host "pgr-dev-xeago-postgres-ha-postgresql-2.pgr-dev-xeago-postgres-ha-postgresql-headless.pgr-dev.svc.cluster.local" (10.0.14.70) and accepting
2020-08-10T13:21:51.334639999Z 	TCP/IP connections on port 5432?
2020-08-10T13:21:51.334645301Z 
2020-08-10T13:21:51.334649924Z [2020-08-10 13:21:51] [DETAIL] attempted to connect using:
2020-08-10T13:21:51.33465477Z   user=repmgr password=snip connect_timeout=5 dbname=repmgr host=pgr-dev-xeago-postgres-ha-postgresql-2.pgr-dev-xeago-postgres-ha-postgresql-headless.pgr-dev.svc.cluster.local port=5432 fallback_application_name=repmgr
2020-08-10T13:21:52.335536071Z [2020-08-10 13:21:52] [ERROR] connection to database failed
2020-08-10T13:21:52.335577157Z [2020-08-10 13:21:52] [DETAIL] 
2020-08-10T13:21:52.335583553Z could not connect to server: Connection refused
2020-08-10T13:21:52.335588265Z 	Is the server running on host "pgr-dev-xeago-postgres-ha-postgresql-2.pgr-dev-xeago-postgres-ha-postgresql-headless.pgr-dev.svc.cluster.local" (10.0.14.70) and accepting
2020-08-10T13:21:52.335594371Z 	TCP/IP connections on port 5432?
2020-08-10T13:21:52.335599019Z 
2020-08-10T13:21:52.33560381Z [2020-08-10 13:21:52] [DETAIL] attempted to connect using:
2020-08-10T13:21:52.335609008Z   user=repmgr password=snip connect_timeout=5 dbname=repmgr host=pgr-dev-xeago-postgres-ha-postgresql-2.pgr-dev-xeago-postgres-ha-postgresql-headless.pgr-dev.svc.cluster.local port=5432 fallback_application_name=repmgr
2020-08-10T13:21:53.336520242Z [2020-08-10 13:21:53] [ERROR] connection to database failed
2020-08-10T13:21:53.336559534Z [2020-08-10 13:21:53] [DETAIL] 
2020-08-10T13:21:53.336566159Z could not connect to server: Connection refused
2020-08-10T13:21:53.33657158Z 	Is the server running on host "pgr-dev-xeago-postgres-ha-postgresql-2.pgr-dev-xeago-postgres-ha-postgresql-headless.pgr-dev.svc.cluster.local" (10.0.14.70) and accepting
2020-08-10T13:21:53.336577926Z 	TCP/IP connections on port 5432?
2020-08-10T13:21:53.336582711Z 
2020-08-10T13:21:53.336587312Z [2020-08-10 13:21:53] [DETAIL] attempted to connect using:
2020-08-10T13:21:53.336592118Z   user=repmgr password=snip connect_timeout=5 dbname=repmgr host=pgr-dev-xeago-postgres-ha-postgresql-2.pgr-dev-xeago-postgres-ha-postgresql-headless.pgr-dev.svc.cluster.local port=5432 fallback_application_name=repmgr
2020-08-10T13:21:54.337432768Z [2020-08-10 13:21:54] [ERROR] connection to database failed
2020-08-10T13:21:54.337466751Z [2020-08-10 13:21:54] [DETAIL] 
2020-08-10T13:21:54.337472743Z could not connect to server: Connection refused
2020-08-10T13:21:54.337476218Z 	Is the server running on host "pgr-dev-xeago-postgres-ha-postgresql-2.pgr-dev-xeago-postgres-ha-postgresql-headless.pgr-dev.svc.cluster.local" (10.0.14.70) and accepting
2020-08-10T13:21:54.337480331Z 	TCP/IP connections on port 5432?
2020-08-10T13:21:54.337483665Z 
2020-08-10T13:21:54.337486733Z [2020-08-10 13:21:54] [DETAIL] attempted to connect using:
2020-08-10T13:21:54.337490189Z   user=repmgr password=snip connect_timeout=5 dbname=repmgr host=pgr-dev-xeago-postgres-ha-postgresql-2.pgr-dev-xeago-postgres-ha-postgresql-headless.pgr-dev.svc.cluster.local port=5432 fallback_application_name=repmgr
2020-08-10T13:21:55.338333492Z [2020-08-10 13:21:55] [ERROR] connection to database failed
2020-08-10T13:21:55.338370837Z [2020-08-10 13:21:55] [DETAIL] 
2020-08-10T13:21:55.338377646Z could not connect to server: Connection refused
2020-08-10T13:21:55.33838271Z 	Is the server running on host "pgr-dev-xeago-postgres-ha-postgresql-2.pgr-dev-xeago-postgres-ha-postgresql-headless.pgr-dev.svc.cluster.local" (10.0.14.70) and accepting
2020-08-10T13:21:55.338388377Z 	TCP/IP connections on port 5432?
2020-08-10T13:21:55.338392794Z 
2020-08-10T13:21:55.338396787Z [2020-08-10 13:21:55] [DETAIL] attempted to connect using:
2020-08-10T13:21:55.338401156Z   user=repmgr password=snip connect_timeout=5 dbname=repmgr host=pgr-dev-xeago-postgres-ha-postgresql-2.pgr-dev-xeago-postgres-ha-postgresql-headless.pgr-dev.svc.cluster.local port=5432 fallback_application_name=repmgr
2020-08-10T13:21:56.339224358Z [2020-08-10 13:21:56] [ERROR] connection to database failed
2020-08-10T13:21:56.33926164Z [2020-08-10 13:21:56] [DETAIL] 
2020-08-10T13:21:56.33926786Z could not connect to server: Connection refused
2020-08-10T13:21:56.339273171Z 	Is the server running on host "pgr-dev-xeago-postgres-ha-postgresql-2.pgr-dev-xeago-postgres-ha-postgresql-headless.pgr-dev.svc.cluster.local" (10.0.14.70) and accepting
2020-08-10T13:21:56.339277953Z 	TCP/IP connections on port 5432?
2020-08-10T13:21:56.339281267Z 
```

```
1001           1  0.1  0.0  15596  6572 ?        Ss   13:10   0:00 /opt/bitnami/repmgr/bin/repmgrd --pid-file=/opt/bitnami/repmgr/tmp/repmgr.pid -f /opt/bitnami/repmgr/conf/repmgr.conf --daemonize=false
1001         427  0.1  0.0      0     0 ?        Z    13:12   0:00 [postgres] <defunct>
1001         428  0.0  0.0      0     0 ?        Zs   13:12   0:00 [postgres] <defunct>
1001        1964  0.0  0.0   2688   900 ?        S    13:21   0:00 sh -c PGPASSWORD=eet5zooNei4Ien2chah9aNge repmgr standby follow -f "/opt/bitnami/repmgr/conf/repmgr.conf" -W --log-level DEBUG --verbose
1001        1965  0.1  0.0  15780  6840 ?        S    13:21   0:00  \_ repmgr standby follow -f /opt/bitnami/repmgr/conf/repmgr.conf -W --log-level DEBUG --verbose
```

Note the `waiting for server to start....postgres: could not access the server configuration file "/bitnami/postgresql/data/postgresql.conf": No such file or directory`, where it fails to start postgres.

**Describe the results you expected:**
A healthy cluster where the replica that changes follower is not killed by the liveliness probe.

**Comment by rafariossaa** [comment]
Hi,
  You said you are using k8s. Are you using the chart ? 
  In order to be able to reproduce the issue under a similar context. Could you give more info about how you did the deployment ?

**Comment by davidsugianto** [comment]
Hi,
I have the same problem, is there a solution for this problem?

```
[2020-09-04 12:36:03] [ERROR] connection to database failed
[2020-09-04 12:36:03] [DETAIL] 
could not connect to server: Connection refused
        Is the server running on host "postgresql-2" (172.22.0.2) and accepting
        TCP/IP connections on port 5432?

[2020-09-04 12:36:03] [DETAIL] attempted to connect using:
  user=replicationmgr password=repmgrpassword connect_timeout=5 dbname=replicationmgr host=postgresql-2 port=5432 fallback_application_name=repmgr
[2020-09-04 12:36:04] [ERROR] connection to database failed
[2020-09-04 12:36:04] [DETAIL] 
could not connect to server: Connection refused
        Is the server running on host "postgresql-2" (172.22.0.2) and accepting
        TCP/IP connections on port 5432?

[2020-09-04 12:36:04] [DETAIL] attempted to connect using:
  user=replicationmgr password=repmgrpassword connect_timeout=5 dbname=replicationmgr host=postgresql-2 port=5432 fallback_application_name=repmgr
[2020-09-04 12:36:05] [ERROR] connection to database failed
[2020-09-04 12:36:05] [DETAIL] 
could not connect to server: Connection refused
        Is the server running on host "postgresql-2" (172.22.0.2) and accepting
        TCP/IP connections on port 5432?

```

**Comment by jxl4650152** [comment]
@xeago @davidsugianto 
I solved the problem by adding following configurations to file `/opt/bitnami/repmgr/conf/repmgr.conf`
```
service_stop_command='/opt/bitnami/postgresql/bin/pg_ctl  -w -D \'/bitnami/postgresql/data\' -o \'--config-file=/opt/bitnami/postgresql/conf/postgresql.conf\' stop'
service_start_command='/opt/bitnami/postgresql/bin/pg_ctl  -w -D \'/bitnami/postgresql/data\' -o \'--config-file=/opt/bitnami/postgresql/conf/postgresql.conf --external_pid_file=/opt/bitnami/postgresql/tmp/postgresql.pid --hba_file=/opt/bitnami/postgresql/conf/pg_hba.conf\' start'
```
Could you give it a try?

**Comment by jxl4650152** [comment]
#22 #33

**Comment by davidsugianto** [comment]
@jxl4650152 
yes thanks my problem has been resolved by adding the config below in `repmgr.conf`
```
service_restart_command='pg_ctl restart -D /bitnami/postgresql/data'
```

**Comment by rafariossaa** [comment]
Hi 
  @jxl4650152, @davidsugianto  Many thanks for providing a workaround.
  I am creating a internal task to include it in the configuration generated and test this further in different scenarios.
  We will come back as soon as we have news.

**Comment by jxl4650152** [comment]
@rafariossaa Thanks, that's great.

**Comment by rafariossaa** [comment]
Hi,
  We did a new relase of the container that moves lock files to persisted volume.
  Could you check if this fix the issue for you ?

**Comment by Enere** [comment]
Hello,

problem still occurs. Tested in pure docker env, 1 master/2 slaves setup with bitnami/postgresql-repmgr:12.4.0-debian-10-r74 : when the master is taken down, the first slave is promoted, the second starts looping.

> We did a new relase of the container that moves lock files to persisted volume.
> Could you check if this fix the issue for you ?
I tried to find your fix in the history, no luck so far. 

Regards,
    Nicolas.

**Comment by rafariossaa** [comment]
Hi @enere 

 I can see this error when enabling debug:
```
pg-2_1  | DEBUG: executing:
pg-2_1  |   /opt/bitnami/postgresql/bin/pg_ctl  -D '/bitnami/postgresql/data' -w -m fast stop 2>/tmp/repmgr_command.F7OmpF
pg-2_1  | 2020-11-06 09:31:04.115 GMT [228] LOG:  received fast shutdown request
pg-2_1  | 2020-11-06 09:31:04.117 GMT [228] LOG:  aborting any active transactions
pg-2_1  | 2020-11-06 09:31:09.121 GMT [231] LOG:  shutting down
pg-2_1  | 2020-11-06 09:31:09.130 GMT [228] LOG:  database system is shut down
pg-2_1  | DEBUG: result of command was 0 (0)
pg-2_1  | DEBUG: local_command(): output returned was:
pg-2_1  | waiting for server to shut down......... done
pg-2_1  | 
pg-2_1  | NOTICE: starting server using "/opt/bitnami/postgresql/bin/pg_ctl  -w -D '/bitnami/postgresql/data' start"
pg-2_1  | DEBUG: executing:
pg-2_1  |   /opt/bitnami/postgresql/bin/pg_ctl  -w -D '/bitnami/postgresql/data' start 2>/tmp/repmgr_command.clBUUh
pg-2_1  | DEBUG: result of command was 141 (36096)
pg-2_1  | DEBUG: local_command(): output returned was:
pg-2_1  | waiting for server to shut down......... done
pg-2_1  | waiting for server to start....postgres: could not access the server configuration file "/bitnami/postgresql/data/postgresql.conf": No such file or directory
```
 So  postgresql is not started hence it at can not be added to the cluster.

  I am creating an internal task to dig into this issue.

**Comment by rafariossaa** [comment]
Hi,
  Just to let you know that we are currently working on this issue.

**Comment by juan131** [comment]
Hi everyone,

I just want to mentioned that we released new versions of the Pgpool and PostgreSQL with Repmgr containers during the weekend that should address this issue.

Could you please give a try to the latest images? Thanks in advance.

**Comment by enere** [comment]
hello, i tried with latest pg 12 version (bitnami/postgresql-repmgr:12.5.0-debian-10-r3) but I am blocked by issue #54 . Will wait for the latest to be fixed :)

**Comment by marcosbc** [comment]
Thanks for sharing, do you still get the same errors as the ones mentioned in this issue? If not please share them.

In any case, I'll update our internal task to mention this seems not to be fully working yet.

**Comment by enere** [comment]
Hi @marcosbc, same issue with postgresql-repmgr:12.5.0-debian-10-r5  : master starts as expected, slaves just stop.

My test setup (no volume set to start from scratch): 
`
version: '2'
services:
  pg-0:
    image: bitnami/postgresql-repmgr:12.5.0-debian-10-r5
    environment:
      REPMGR_PARTNER_NODES: pg-0,pg-1,pg-3
      REPMGR_NODE_NAME: pg-0
      REPMGR_NODE_NETWORK_NAME: pg-0
      REPMGR_PRIMARY_HOST: pg-0
      REPMGR_PASSWORD: repmgrpass
      POSTGRESQL_POSTGRES_PASSWORD: adminpassword
      POSTGRESQL_USERNAME: customuser
      POSTGRESQL_PASSWORD: custompassword
      POSTGRESQL_DATABASE: customdatabase
      REPMGR_USERNAME: repmgr
      REPMGR_NODE_PRIORITY: '200'
  pg-1:
    image: bitnami/postgresql-repmgr:12.5.0-debian-10-r5
    environment:
      REPMGR_PARTNER_NODES: pg-0,pg-1,pg-2
      REPMGR_NODE_NAME: pg-1
      REPMGR_NODE_NETWORK_NAME: pg-1
      REPMGR_PRIMARY_HOST: pg-0
      REPMGR_PASSWORD: repmgrpass
      POSTGRESQL_POSTGRES_PASSWORD: adminpassword
      POSTGRESQL_USERNAME: customuser
      POSTGRESQL_PASSWORD: custompassword
      POSTGRESQL_DATABASE: customdatabase
      REPMGR_USERNAME: repmgr
  pg-2:
    image: bitnami/postgresql-repmgr:12.5.0-debian-10-r5
    environment:
      REPMGR_PARTNER_NODES: pg-0,pg-1,pg-2
      REPMGR_NODE_NAME: pg-2
      REPMGR_NODE_NETWORK_NAME: pg-2
      REPMGR_PRIMARY_HOST: pg-0
      REPMGR_PASSWORD: repmgrpass
      POSTGRESQL_POSTGRES_PASSWORD: adminpassword
      POSTGRESQL_USERNAME: customuser
      POSTGRESQL_PASSWORD: custompassword
      POSTGRESQL_DATABASE: customdatabase
      REPMGR_USERNAME: repmgr
`
Master logs (pg-0):
`
19/11/2020 11:53:41postgresql-repmgr 10:53:41.23 
19/11/2020 11:53:41postgresql-repmgr 10:53:41.24 Welcome to the Bitnami postgresql-repmgr container
19/11/2020 11:53:41postgresql-repmgr 10:53:41.24 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-postgresql-repmgr
19/11/2020 11:53:41postgresql-repmgr 10:53:41.24 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-postgresql-repmgr/issues
19/11/2020 11:53:41postgresql-repmgr 10:53:41.24 
19/11/2020 11:53:41postgresql-repmgr 10:53:41.28 INFO  ==> ** Starting PostgreSQL with Replication Manager setup **
19/11/2020 11:53:41postgresql-repmgr 10:53:41.32 INFO  ==> Validating settings in REPMGR_* env vars...
19/11/2020 11:53:41postgresql-repmgr 10:53:41.32 INFO  ==> Validating settings in POSTGRESQL_* env vars..
19/11/2020 11:53:41postgresql-repmgr 10:53:41.33 INFO  ==> Querying all partner nodes for common upstream node...
19/11/2020 11:53:44postgresql-repmgr 10:53:44.48 INFO  ==> There are no nodes with primary role. Assuming the primary role...
19/11/2020 11:53:44postgresql-repmgr 10:53:44.49 INFO  ==> Preparing PostgreSQL configuration...
19/11/2020 11:53:44postgresql-repmgr 10:53:44.50 INFO  ==> postgresql.conf file not detected. Generating it...
19/11/2020 11:53:44postgresql-repmgr 10:53:44.65 INFO  ==> Preparing repmgr configuration...
19/11/2020 11:53:44postgresql-repmgr 10:53:44.66 INFO  ==> Initializing Repmgr...
19/11/2020 11:53:44postgresql-repmgr 10:53:44.67 INFO  ==> Initializing PostgreSQL database...
19/11/2020 11:53:44postgresql-repmgr 10:53:44.68 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/postgresql.conf detected
19/11/2020 11:53:44postgresql-repmgr 10:53:44.68 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/pg_hba.conf detected
19/11/2020 11:53:46postgresql-repmgr 10:53:46.54 INFO  ==> Starting PostgreSQL in background...
19/11/2020 11:53:46postgresql-repmgr 10:53:46.83 INFO  ==> Changing password of postgres
19/11/2020 11:53:46postgresql-repmgr 10:53:46.86 INFO  ==> Creating user customuser
19/11/2020 11:53:46postgresql-repmgr 10:53:46.88 INFO  ==> Granting access to "customuser" to the database "customdatabase"
19/11/2020 11:53:46postgresql-repmgr 10:53:46.90 INFO  ==> Creating replication user repmgr
19/11/2020 11:53:46postgresql-repmgr 10:53:46.96 INFO  ==> Configuring replication parameters
19/11/2020 11:53:47postgresql-repmgr 10:53:47.04 INFO  ==> Configuring fsync
19/11/2020 11:53:47postgresql-repmgr 10:53:47.05 INFO  ==> Starting PostgreSQL in background...
19/11/2020 11:53:47postgresql-repmgr 10:53:47.06 INFO  ==> Creating repmgr user: repmgr
19/11/2020 11:53:47postgresql-repmgr 10:53:47.11 INFO  ==> Creating repmgr database: repmgr
19/11/2020 11:53:47postgresql-repmgr 10:53:47.27 INFO  ==> Stopping PostgreSQL...
19/11/2020 11:53:47waiting for server to shut down.... done
19/11/2020 11:53:47server stopped
19/11/2020 11:53:47postgresql-repmgr 10:53:47.49 INFO  ==> Starting PostgreSQL in background...
19/11/2020 11:53:47postgresql-repmgr 10:53:47.64 INFO  ==> Registering Primary...
19/11/2020 11:53:47postgresql-repmgr 10:53:47.83 INFO  ==> Loading custom scripts...
19/11/2020 11:53:47postgresql-repmgr 10:53:47.84 INFO  ==> Stopping PostgreSQL...
19/11/2020 11:53:48waiting for server to shut down.... done
19/11/2020 11:53:48server stopped
19/11/2020 11:53:48postgresql-repmgr 10:53:48.05 INFO  ==> ** PostgreSQL with Replication Manager setup finished! **
19/11/2020 11:53:48
19/11/2020 11:53:48postgresql-repmgr 10:53:48.10 INFO  ==> Starting PostgreSQL in background...
19/11/2020 11:53:48waiting for server to start....2020-11-19 10:53:48.146 GMT [263] LOG:  pgaudit extension initialized
19/11/2020 11:53:482020-11-19 10:53:48.146 GMT [263] LOG:  starting PostgreSQL 12.5 on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
19/11/2020 11:53:482020-11-19 10:53:48.146 GMT [263] LOG:  listening on IPv4 address "0.0.0.0", port 5432
19/11/2020 11:53:482020-11-19 10:53:48.146 GMT [263] LOG:  listening on IPv6 address "::", port 5432
19/11/2020 11:53:482020-11-19 10:53:48.161 GMT [263] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
19/11/2020 11:53:482020-11-19 10:53:48.187 GMT [263] LOG:  redirecting log output to logging collector process
19/11/2020 11:53:482020-11-19 10:53:48.187 GMT [263] HINT:  Future log output will appear in directory "/opt/bitnami/postgresql/logs".
19/11/2020 11:53:482020-11-19 10:53:48.199 GMT [265] LOG:  database system was shut down at 2020-11-19 10:53:47 GMT
19/11/2020 11:53:482020-11-19 10:53:48.205 GMT [263] LOG:  database system is ready to accept connections
19/11/2020 11:53:48 done
19/11/2020 11:53:48server started
19/11/2020 11:53:48postgresql-repmgr 10:53:48.24 INFO  ==> ** Starting repmgrd **
19/11/2020 11:53:48[2020-11-19 10:53:48] [NOTICE] repmgrd (repmgrd 5.2.0) starting up
19/11/2020 11:53:48INFO:  set_repmgrd_pid(): provided pidfile is /opt/bitnami/repmgr/tmp/repmgr.pid
19/11/2020 11:53:48[2020-11-19 10:53:48] [NOTICE] starting monitoring of node "pg-0" (ID: 1000)
19/11/2020 11:53:48[2020-11-19 10:53:48] [NOTICE] monitoring cluster primary "pg-0" (ID: 1000)
19/11/2020 11:54:232020-11-19 10:54:23.944 GMT [308] LOG:  could not receive data from client: Connection reset by peer
19/11/2020 11:54:54[2020-11-19 10:54:54] [NOTICE] new standby "pg-1" (ID: 1001) has connected
19/11/2020 11:55:00[2020-11-19 10:55:00] [NOTICE] new standby "pg-2" (ID: 1002) has connected
`

first slave (pg-1) and second slave (pg-2) : 
`
19/11/2020 11:54:37postgresql-repmgr 10:54:37.17 
19/11/2020 11:54:37postgresql-repmgr 10:54:37.17 Welcome to the Bitnami postgresql-repmgr container
19/11/2020 11:54:37postgresql-repmgr 10:54:37.17 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-postgresql-repmgr
19/11/2020 11:54:37postgresql-repmgr 10:54:37.18 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-postgresql-repmgr/issues
19/11/2020 11:54:37postgresql-repmgr 10:54:37.18 
19/11/2020 11:54:37postgresql-repmgr 10:54:37.22 INFO  ==> ** Starting PostgreSQL with Replication Manager setup **
19/11/2020 11:54:37postgresql-repmgr 10:54:37.26 INFO  ==> Validating settings in REPMGR_* env vars...
19/11/2020 11:54:37postgresql-repmgr 10:54:37.26 INFO  ==> Validating settings in POSTGRESQL_* env vars..
19/11/2020 11:54:37postgresql-repmgr 10:54:37.27 INFO  ==> Querying all partner nodes for common upstream node...
19/11/2020 11:54:47postgresql-repmgr 10:54:47.49 INFO  ==> Auto-detected primary node: 'pg-0:5432'
19/11/2020 11:54:47postgresql-repmgr 10:54:47.50 INFO  ==> Preparing PostgreSQL configuration...
19/11/2020 11:54:47postgresql-repmgr 10:54:47.51 INFO  ==> postgresql.conf file not detected. Generating it...
19/11/2020 11:54:47postgresql-repmgr 10:54:47.69 INFO  ==> Preparing repmgr configuration...
19/11/2020 11:54:47postgresql-repmgr 10:54:47.70 INFO  ==> Initializing Repmgr...
19/11/2020 11:54:47postgresql-repmgr 10:54:47.72 INFO  ==> Waiting for primary node...
19/11/2020 11:54:47postgresql-repmgr 10:54:47.81 INFO  ==> Cloning data from primary node...
19/11/2020 11:54:50postgresql-repmgr 10:54:50.11 INFO  ==> Initializing PostgreSQL database...
19/11/2020 11:54:50postgresql-repmgr 10:54:50.12 INFO  ==> Cleaning stale /bitnami/postgresql/data/standby.signal file
19/11/2020 11:54:50postgresql-repmgr 10:54:50.14 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/postgresql.conf detected
19/11/2020 11:54:50postgresql-repmgr 10:54:50.14 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/pg_hba.conf detected
19/11/2020 11:54:50postgresql-repmgr 10:54:50.19 INFO  ==> Deploying PostgreSQL with persisted data...
19/11/2020 11:54:50postgresql-repmgr 10:54:50.24 INFO  ==> Configuring replication parameters
19/11/2020 11:54:50postgresql-repmgr 10:54:50.30 INFO  ==> Configuring fsync
19/11/2020 11:54:50postgresql-repmgr 10:54:50.33 INFO  ==> Setting up streaming replication slave...
19/11/2020 11:54:50postgresql-repmgr 10:54:50.37 INFO  ==> Starting PostgreSQL in background...
19/11/2020 11:54:50postgresql-repmgr 10:54:50.75 INFO  ==> Unregistering standby node...
19/11/2020 11:54:50postgresql-repmgr 10:54:50.85 INFO  ==> Registering Standby node...
19/11/2020 11:54:51postgresql-repmgr 10:54:51.09 INFO  ==> Running standby follow...
19/11/2020 11:54:51postgresql-repmgr 10:54:51.39 INFO  ==> Stopping PostgreSQL...
19/11/2020 11:54:51waiting for server to shut down.... done
19/11/2020 11:54:51server stopped
`
Regards,
    Nicolas.

**Comment by enere** [comment]
Hi @marcosbc, same issue with postgresql-repmgr:12.5.0-debian-10-r5  : master starts as expected, slaves just stop.

My test setup (no volume set to start from scratch): 
```docker
version: '2'
services:
  pg-0:
    image: bitnami/postgresql-repmgr:12.5.0-debian-10-r5
    environment:
      REPMGR_PARTNER_NODES: pg-0,pg-1,pg-3
      REPMGR_NODE_NAME: pg-0
      REPMGR_NODE_NETWORK_NAME: pg-0
      REPMGR_PRIMARY_HOST: pg-0
      REPMGR_PASSWORD: repmgrpass
      POSTGRESQL_POSTGRES_PASSWORD: adminpassword
      POSTGRESQL_USERNAME: customuser
      POSTGRESQL_PASSWORD: custompassword
      POSTGRESQL_DATABASE: customdatabase
      REPMGR_USERNAME: repmgr
      REPMGR_NODE_PRIORITY: '200'
  pg-1:
    image: bitnami/postgresql-repmgr:12.5.0-debian-10-r5
    environment:
      REPMGR_PARTNER_NODES: pg-0,pg-1,pg-2
      REPMGR_NODE_NAME: pg-1
      REPMGR_NODE_NETWORK_NAME: pg-1
      REPMGR_PRIMARY_HOST: pg-0
      REPMGR_PASSWORD: repmgrpass
      POSTGRESQL_POSTGRES_PASSWORD: adminpassword
      POSTGRESQL_USERNAME: customuser
      POSTGRESQL_PASSWORD: custompassword
      POSTGRESQL_DATABASE: customdatabase
      REPMGR_USERNAME: repmgr
  pg-2:
    image: bitnami/postgresql-repmgr:12.5.0-debian-10-r5
    environment:
      REPMGR_PARTNER_NODES: pg-0,pg-1,pg-2
      REPMGR_NODE_NAME: pg-2
      REPMGR_NODE_NETWORK_NAME: pg-2
      REPMGR_PRIMARY_HOST: pg-0
      REPMGR_PASSWORD: repmgrpass
      POSTGRESQL_POSTGRES_PASSWORD: adminpassword
      POSTGRESQL_USERNAME: customuser
      POSTGRESQL_PASSWORD: custompassword
      POSTGRESQL_DATABASE: customdatabase
      REPMGR_USERNAME: repmgr
```
Master logs (pg-0):
```
19/11/2020 11:53:41postgresql-repmgr 10:53:41.23 
19/11/2020 11:53:41postgresql-repmgr 10:53:41.24 Welcome to the Bitnami postgresql-repmgr container
19/11/2020 11:53:41postgresql-repmgr 10:53:41.24 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-postgresql-repmgr
19/11/2020 11:53:41postgresql-repmgr 10:53:41.24 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-postgresql-repmgr/issues
19/11/2020 11:53:41postgresql-repmgr 10:53:41.24 
19/11/2020 11:53:41postgresql-repmgr 10:53:41.28 INFO  ==> ** Starting PostgreSQL with Replication Manager setup **
19/11/2020 11:53:41postgresql-repmgr 10:53:41.32 INFO  ==> Validating settings in REPMGR_* env vars...
19/11/2020 11:53:41postgresql-repmgr 10:53:41.32 INFO  ==> Validating settings in POSTGRESQL_* env vars..
19/11/2020 11:53:41postgresql-repmgr 10:53:41.33 INFO  ==> Querying all partner nodes for common upstream node...
19/11/2020 11:53:44postgresql-repmgr 10:53:44.48 INFO  ==> There are no nodes with primary role. Assuming the primary role...
19/11/2020 11:53:44postgresql-repmgr 10:53:44.49 INFO  ==> Preparing PostgreSQL configuration...
19/11/2020 11:53:44postgresql-repmgr 10:53:44.50 INFO  ==> postgresql.conf file not detected. Generating it...
19/11/2020 11:53:44postgresql-repmgr 10:53:44.65 INFO  ==> Preparing repmgr configuration...
19/11/2020 11:53:44postgresql-repmgr 10:53:44.66 INFO  ==> Initializing Repmgr...
19/11/2020 11:53:44postgresql-repmgr 10:53:44.67 INFO  ==> Initializing PostgreSQL database...
19/11/2020 11:53:44postgresql-repmgr 10:53:44.68 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/postgresql.conf detected
19/11/2020 11:53:44postgresql-repmgr 10:53:44.68 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/pg_hba.conf detected
19/11/2020 11:53:46postgresql-repmgr 10:53:46.54 INFO  ==> Starting PostgreSQL in background...
19/11/2020 11:53:46postgresql-repmgr 10:53:46.83 INFO  ==> Changing password of postgres
19/11/2020 11:53:46postgresql-repmgr 10:53:46.86 INFO  ==> Creating user customuser
19/11/2020 11:53:46postgresql-repmgr 10:53:46.88 INFO  ==> Granting access to "customuser" to the database "customdatabase"
19/11/2020 11:53:46postgresql-repmgr 10:53:46.90 INFO  ==> Creating replication user repmgr
19/11/2020 11:53:46postgresql-repmgr 10:53:46.96 INFO  ==> Configuring replication parameters
19/11/2020 11:53:47postgresql-repmgr 10:53:47.04 INFO  ==> Configuring fsync
19/11/2020 11:53:47postgresql-repmgr 10:53:47.05 INFO  ==> Starting PostgreSQL in background...
19/11/2020 11:53:47postgresql-repmgr 10:53:47.06 INFO  ==> Creating repmgr user: repmgr
19/11/2020 11:53:47postgresql-repmgr 10:53:47.11 INFO  ==> Creating repmgr database: repmgr
19/11/2020 11:53:47postgresql-repmgr 10:53:47.27 INFO  ==> Stopping PostgreSQL...
19/11/2020 11:53:47waiting for server to shut down.... done
19/11/2020 11:53:47server stopped
19/11/2020 11:53:47postgresql-repmgr 10:53:47.49 INFO  ==> Starting PostgreSQL in background...
19/11/2020 11:53:47postgresql-repmgr 10:53:47.64 INFO  ==> Registering Primary...
19/11/2020 11:53:47postgresql-repmgr 10:53:47.83 INFO  ==> Loading custom scripts...
19/11/2020 11:53:47postgresql-repmgr 10:53:47.84 INFO  ==> Stopping PostgreSQL...
19/11/2020 11:53:48waiting for server to shut down.... done
19/11/2020 11:53:48server stopped
19/11/2020 11:53:48postgresql-repmgr 10:53:48.05 INFO  ==> ** PostgreSQL with Replication Manager setup finished! **
19/11/2020 11:53:48
19/11/2020 11:53:48postgresql-repmgr 10:53:48.10 INFO  ==> Starting PostgreSQL in background...
19/11/2020 11:53:48waiting for server to start....2020-11-19 10:53:48.146 GMT [263] LOG:  pgaudit extension initialized
19/11/2020 11:53:482020-11-19 10:53:48.146 GMT [263] LOG:  starting PostgreSQL 12.5 on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
19/11/2020 11:53:482020-11-19 10:53:48.146 GMT [263] LOG:  listening on IPv4 address "0.0.0.0", port 5432
19/11/2020 11:53:482020-11-19 10:53:48.146 GMT [263] LOG:  listening on IPv6 address "::", port 5432
19/11/2020 11:53:482020-11-19 10:53:48.161 GMT [263] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
19/11/2020 11:53:482020-11-19 10:53:48.187 GMT [263] LOG:  redirecting log output to logging collector process
19/11/2020 11:53:482020-11-19 10:53:48.187 GMT [263] HINT:  Future log output will appear in directory "/opt/bitnami/postgresql/logs".
19/11/2020 11:53:482020-11-19 10:53:48.199 GMT [265] LOG:  database system was shut down at 2020-11-19 10:53:47 GMT
19/11/2020 11:53:482020-11-19 10:53:48.205 GMT [263] LOG:  database system is ready to accept connections
19/11/2020 11:53:48 done
19/11/2020 11:53:48server started
19/11/2020 11:53:48postgresql-repmgr 10:53:48.24 INFO  ==> ** Starting repmgrd **
19/11/2020 11:53:48[2020-11-19 10:53:48] [NOTICE] repmgrd (repmgrd 5.2.0) starting up
19/11/2020 11:53:48INFO:  set_repmgrd_pid(): provided pidfile is /opt/bitnami/repmgr/tmp/repmgr.pid
19/11/2020 11:53:48[2020-11-19 10:53:48] [NOTICE] starting monitoring of node "pg-0" (ID: 1000)
19/11/2020 11:53:48[2020-11-19 10:53:48] [NOTICE] monitoring cluster primary "pg-0" (ID: 1000)
19/11/2020 11:54:232020-11-19 10:54:23.944 GMT [308] LOG:  could not receive data from client: Connection reset by peer
19/11/2020 11:54:54[2020-11-19 10:54:54] [NOTICE] new standby "pg-1" (ID: 1001) has connected
19/11/2020 11:55:00[2020-11-19 10:55:00] [NOTICE] new standby "pg-2" (ID: 1002) has connected
```

first slave (pg-1) and second slave (pg-2) : 
```
19/11/2020 11:54:37postgresql-repmgr 10:54:37.17 
19/11/2020 11:54:37postgresql-repmgr 10:54:37.17 Welcome to the Bitnami postgresql-repmgr container
19/11/2020 11:54:37postgresql-repmgr 10:54:37.17 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-postgresql-repmgr
19/11/2020 11:54:37postgresql-repmgr 10:54:37.18 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-postgresql-repmgr/issues
19/11/2020 11:54:37postgresql-repmgr 10:54:37.18 
19/11/2020 11:54:37postgresql-repmgr 10:54:37.22 INFO  ==> ** Starting PostgreSQL with Replication Manager setup **
19/11/2020 11:54:37postgresql-repmgr 10:54:37.26 INFO  ==> Validating settings in REPMGR_* env vars...
19/11/2020 11:54:37postgresql-repmgr 10:54:37.26 INFO  ==> Validating settings in POSTGRESQL_* env vars..
19/11/2020 11:54:37postgresql-repmgr 10:54:37.27 INFO  ==> Querying all partner nodes for common upstream node...
19/11/2020 11:54:47postgresql-repmgr 10:54:47.49 INFO  ==> Auto-detected primary node: 'pg-0:5432'
19/11/2020 11:54:47postgresql-repmgr 10:54:47.50 INFO  ==> Preparing PostgreSQL configuration...
19/11/2020 11:54:47postgresql-repmgr 10:54:47.51 INFO  ==> postgresql.conf file not detected. Generating it...
19/11/2020 11:54:47postgresql-repmgr 10:54:47.69 INFO  ==> Preparing repmgr configuration...
19/11/2020 11:54:47postgresql-repmgr 10:54:47.70 INFO  ==> Initializing Repmgr...
19/11/2020 11:54:47postgresql-repmgr 10:54:47.72 INFO  ==> Waiting for primary node...
19/11/2020 11:54:47postgresql-repmgr 10:54:47.81 INFO  ==> Cloning data from primary node...
19/11/2020 11:54:50postgresql-repmgr 10:54:50.11 INFO  ==> Initializing PostgreSQL database...
19/11/2020 11:54:50postgresql-repmgr 10:54:50.12 INFO  ==> Cleaning stale /bitnami/postgresql/data/standby.signal file
19/11/2020 11:54:50postgresql-repmgr 10:54:50.14 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/postgresql.conf detected
19/11/2020 11:54:50postgresql-repmgr 10:54:50.14 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/pg_hba.conf detected
19/11/2020 11:54:50postgresql-repmgr 10:54:50.19 INFO  ==> Deploying PostgreSQL with persisted data...
19/11/2020 11:54:50postgresql-repmgr 10:54:50.24 INFO  ==> Configuring replication parameters
19/11/2020 11:54:50postgresql-repmgr 10:54:50.30 INFO  ==> Configuring fsync
19/11/2020 11:54:50postgresql-repmgr 10:54:50.33 INFO  ==> Setting up streaming replication slave...
19/11/2020 11:54:50postgresql-repmgr 10:54:50.37 INFO  ==> Starting PostgreSQL in background...
19/11/2020 11:54:50postgresql-repmgr 10:54:50.75 INFO  ==> Unregistering standby node...
19/11/2020 11:54:50postgresql-repmgr 10:54:50.85 INFO  ==> Registering Standby node...
19/11/2020 11:54:51postgresql-repmgr 10:54:51.09 INFO  ==> Running standby follow...
19/11/2020 11:54:51postgresql-repmgr 10:54:51.39 INFO  ==> Stopping PostgreSQL...
19/11/2020 11:54:51waiting for server to shut down.... done
19/11/2020 11:54:51server stopped
```
Regards,
    Nicolas.

**Comment by marcosbc** [comment]
Thanks! I've added the details to our internal task, to note that it is still not fixed. We'll keep this issue updated with any progress.

**Comment by enere** [comment]
Hi @marcosbc, I did some tests today using 12.5.0-debian-10-r18, issue seems to be gone, great !
* my setup: 3 nodes (one master, two slaves) ;
* when master is stopped, one of the remaining slave is promoted *and* the other one follows the new master, as expected ;
* when old master is back online, it starts to follow the new master, as expected ;

I still have to find how to promote the old master without killing the new one : is there a script somewhere ?
I will do more real-condition tests in the next few days, will keep you informed.

**Comment by rafariossaa** [comment]
Hi @enere ,
  I am happy the issue is finally fixed.
  You should be able to promote a node by connecting to the node and issuing the repmgr commands.
  Don't hesitate to come back and share your findings.

**Comment by enere** [comment]
> You should be able to promote a node by connecting to the node and issuing the repmgr commands.

It seems that it require ssh to be available and set-up (see #51). I will see to integrate @sbJonas solution in our own images.

**Comment by rafariossaa** [comment]
Hi,
  I am happy you were able to find a workaround.
  Thanks for sharing.

**Comment by xeago** [comment]
@rafariossaa apologies, I got occupied with other tasks as we deemed this to have too many issues before taking it to production.
We run the chart on k8s, with a mostly standard setup.

Even with the changes proposed in this thread; I believe this to still have problems, perhaps better addressed in another issue: Over time, the master node may drift to a node that is not the zeroth in the statefulset. Given a failure of some kind in which each replica is unavailable, this chart does not allow for recovery of the cluster.

**Comment by rafariossaa** [comment]
Hi @xeago ,
  Sure, please don't hesitate to open a new issue.

**Comment by rafariossaa** [comment]
I am proceeding to close this issue.
If some other case appears, please don't hesitate to reopen this or create a new one.

**Comment by xeago** [comment]
Hi @rafariossaa, apologies, I'm no longer working in this domain. I've just verified on a k8s cluster running locally that a simultaneous shutdown of the 2 replicas and the primary results in an unhealthy cluster, if the primary was not the 0th node. This occurs due to the assumption of elected-primary-in-db being observable in the environment (at a future time) when the nodes boot up one by one; that fails to hold up in this situation.

That's all the time I sadly have available to me. With this message and the previous, I've also shared all the knowledge available to me.

**Comment by rafariossaa** [comment]
Hi, 
  Thanks you very much for you time and providing feedback.
  I am adding this notes to our current internal task.

**Comment by rafariossaa** [comment]
I tried to reproduce the issue with the current version of the docker image and I am no longer able to reproduce the issue.
Please, don't hesitate to reopen it if needed.
I am closing the issue.
